src/Impostor.Server/Net/Inner/Objects/GameManager/Logic/GameLogicComponent.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Impostor.Api.Net.Inner;
+using Impostor.Api.Net.Inner.Objects;
+
+namespace Impostor.Server.Net.Inner.Objects.GameManager.Logic;
+
+internal abstract class GameLogicComponent
+{
+    public virtual void OnPlayerDisconnect(IInnerPlayerControl pc)
+    {
+    }
+
+    public virtual void HandleRPC(RpcCalls callId, IMessageReader reader)
+    {
+    }
+
+    public abstract bool Serialize(IMessageWriter writer, bool initialState);
```

Comment:
Most implementations are empty anyway, make this virtual?

Reply:
I don't think we use any Serialize function in Impostor, should be okay to make this virtual to remove unnecessary code.